glean-core/src/metrics/labeled.rs

```diff
@@ -86,6 +89,32 @@ pub struct LabeledMetric<T> {
     label_map: Mutex<HashMap<String, Arc<T>>>,
 }
 
+impl<T: MallocSizeOf> ::malloc_size_of::MallocSizeOf for LabeledMetric<T> {
+    fn size_of(&self, ops: &mut malloc_size_of::MallocSizeOfOps) -> usize {
+        let map = self.label_map.lock().unwrap();
+
+        // Copy of `MallocShallowSizeOf` implementation for `HashMap<K, V>`.
+        // Note: An instantiated submetric is behind an `Arc`.
+        // `size_of` should only be called on the main thread to avoid double-counting.
```

Comment:
This sounds like a very Firefox-specific concern. Is there anything we can do about this in a cross-product way?